bin/main

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env ruby
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    entry
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    entry
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    arr = []
+    while i < size
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |e| return false unless yield(e) }
+    end
+    if arg.nil?
+      my_each { |e| return false unless e }
+      return true
+    end
+    my_each { |e| return false unless check_validity(e, arg) }
+    true
+  end
```

Comment:
Method my_all? failed the test with Regex expressions

enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)